test/index.js

```diff
@@ -359,6 +361,51 @@ describe('Consumer', function () {
 
       consumer.start();
     });
+
+    it('terminate message visibility timeout on processing error', function(done) {
+      handleMessage.yields(new Error('Processing error'));
+
+      // Turn on terminate visibility timeout option
+      consumer.terminateVisibilityTimeout = true;
```

Comment:
Thanks for adding some tests. I think we need one more test to ensure that `sqs.changeMessageVisibility` is not called when `terminateVisibilityTimeout` is set to false.